lib/autoload/active_record/associations/preloader/manual_has_many_through.rb

```diff
@@ -0,0 +1,5 @@
+class ActiveRecord::Associations::Preloader
+  class ManualHasManyThrough < HasMany
```

Comment:
It's actually not hard, the logic inside the preloader is to deal with the intermediate through association.